Source/RxSwift/ObservableType+Weak.swift

```diff
@@ -83,6 +119,33 @@ extension ObservableType {
 
 		return Disposables.create(self.asObservable().subscribe(observer), disposable)
 	}
+    
+    /**
+     Subscribes an element handler, an error handler, a completion handler and disposed handler to an observable sequence.
+     
+     - parameter weak: The object that we want to have a safe and unretained reference on.
+     - parameter onNext: Action to invoke for each element in the observable sequence.
+     - parameter onError: Action to invoke upon errored termination of the observable sequence..
+     - parameter onCompleted: Action to invoke upon graceful termination of the observable sequence..
+     - parameter onDisposed: FAction to invoke upon any type of termination of sequence (if the sequence has gracefully completed, errored, or if the generation is canceled by disposing subscription).
+     - returns: Subscription object used to unsubscribe from the observable sequence.
+     */
+    public func subscribe<A: AnyObject>(weak obj: A,
+                                        onNext: ((A, E) -> Void)? = nil,
+                                        onError: ((A, Error) -> Void)? = nil,
+                                        onCompleted: ((A) -> Void)? = nil,
+                                        onDisposed: ((A) -> Void)? = nil) -> Disposable {
+        let curriedOnNext: ((A) -> (Self.E) -> Void) = { (strongObj: A) in { (value: E) in onNext?(strongObj, value) } }
```

Comment:
very cool! :) 
Currying always sort of looks like a nightmare to me but here this is used very smartly ;) 

It's a bit confusing though. Wonder if there's any good way to break it up a bit, by at least moving the curried blocks to their own lines. What do you think ?

Reply:
I'm not sure I understand what you mean by "moving the curried blocks to their own lines".

Reply:
Nevermind, re-read this. It's fine :)